compiler/src/Language/Bond/Codegen/Java/Util.hs

```diff
@@ -4,110 +4,69 @@
-            if value > intMax || value < intMin
-            then [lt|#{value}L|]
-            else [lt|#{value}|]
-    explicitDefault (DefaultFloat x) = floatLiteral type_ x
-      where
-        floatLiteral BT_Float y = [lt|#{y}f|]
-        floatLiteral BT_Double y = [lt|#{y}|]
-        floatLiteral _ _ = error "Java:Float:defaultValue/floatLiteral: impossible happened."
-    explicitDefault (DefaultBool True) = "true"
-    explicitDefault (DefaultBool False) = "false"
-    explicitDefault (DefaultString x) = strLiteral type_ x
-      where
-        strLiteral BT_String value = [lt|"#{value}"|]
-        strLiteral BT_WString value = [lt|"#{value}"|]
-        strLiteral _ _ = error "Java:Str:defaultValue/floatLiteral: impossible happened."
-    explicitDefault DefaultNothing = [lt|null|]
-    explicitDefault (DefaultEnum x) = [lt|#{getTypeName java type_}.#{x}|]
+-- returns the qualified name of Modifier constant
+modifierConstantName :: Modifier -> Text
+modifierConstantName Optional = [lt|com.microsoft.bond.Modifier.Optional|]
+modifierConstantName Required = [lt|com.microsoft.bond.Modifier.Required|]
+modifierConstantName RequiredOptional = [lt|com.microsoft.bond.Modifier.RequiredOptional|]
 
-qualifiedName :: MappingContext -> Declaration -> String
-qualifiedName java s@Struct {..}  = intercalate "." $ getDeclNamespace java s ++ [declName]
-qualifiedName java e@Enum {..}  = intercalate "." $ getDeclNamespace java e ++ [declName]
-qualifiedName _ t = error $ "invalid declaration type for qualifiedName: " ++ (show t)
+-- returns a value indicating whether a type is a non-enum Bond primitive type
+isPrimitiveNonEnumBondType :: Type -> Bool
+isPrimitiveNonEnumBondType BT_Int8 = True
+isPrimitiveNonEnumBondType BT_Int16 = True
+isPrimitiveNonEnumBondType BT_Int32 = True
+isPrimitiveNonEnumBondType BT_Int64 = True
+isPrimitiveNonEnumBondType BT_UInt8 = True
+isPrimitiveNonEnumBondType BT_UInt16 = True
+isPrimitiveNonEnumBondType BT_UInt32 = True
+isPrimitiveNonEnumBondType BT_UInt64 = True
+isPrimitiveNonEnumBondType BT_Float = True
+isPrimitiveNonEnumBondType BT_Double = True
+isPrimitiveNonEnumBondType BT_Bool = True
+isPrimitiveNonEnumBondType BT_String = True
+isPrimitiveNonEnumBondType BT_WString = True
+isPrimitiveNonEnumBondType _ = False
 
-generatedClassAnnotations :: Text
-generatedClassAnnotations = [lt|@javax.annotation.Generated("gbc")|]
+-- returns a value indicating whether a type is a Bond primitive type or enum
+isPrimitiveBondType :: Type -> Bool
+isPrimitiveBondType (BT_UserDefined Enum {..} _) = True
+isPrimitiveBondType t = isPrimitiveNonEnumBondType t
+
+-- returns a value indicating whether a type is a generic struct type with generic type parameters
+isGenericBondStructType :: Type -> Bool
+isGenericBondStructType (BT_UserDefined Struct {..} _) = not (null declParams)
+isGenericBondStructType _ = False
+
+-- takes a predicate and two texts and outputs one based on predicate
+ifThenElse :: Bool -> Text -> Text -> Text
```

Comment:
It's in Codegen/Java/Util.hs, so I'm fine leaving it as Text. If we move this some where more general, it should be polymorphic.